src/head_align_tail_dist.py

```diff
@@ -58,6 +58,12 @@ def head_align_tail(prefix, alnm_ext, mode):
         total_ref_length = []  # total length of reference
         prefix = prefix[:-14]
 
+        dict_genome_alnm_info = {}
+        in_sam_file_genome = pysam.AlignmentFile(prefix + "_genome_primary.bam", 'rb')
+        for alnm_temp in in_sam_file_genome.fetch(until_eof=True):
+            read_temp = alnm_temp.query_name
+            dict_genome_alnm_info[read_temp] = alnm_temp
```

Comment:
Two problems here:
1. Split alignments are not accounted for here. Any subsequent alignment records for the same read ID will always overwrite the previous one stored in the dictionary.
2. Memory usage would be an issue because all alignments are stored in the dictionary. I would suggest extracting only the head/tail lengths and store those as a tuple in the dictionary.

Reply:
Thanks for reviewing this and providing feedbacks. I edited the script so that:

1. For any read ID, it stores the information for its subsequent alignment records
2. It only stores the length information for head and tail regions as tuples inside the dictionary, reducing the memory usage.